riscv/src/compiler.rs

```diff
@@ -1264,7 +1265,7 @@ fn process_instruction(instr: &str, args: &[Argument], coprocessors: &CoProcesso
                 format!("tmp3 <== and({rs}, 0xffff);"),
                 "tmp3 <== shl(tmp3, 8 * tmp2);".to_string(),
                 "tmp1 <== or(tmp1, tmp3);".to_string(),
-                "mstore tmp1;".to_string(),
+                "mstore addr, tmp1;".to_string(),
```

Comment:
This case and the two below still require the address to be saved somewhere at first. When we remove `addr` completely we can use something else. If there are no registers available we can rename `addr` to `tmp4`.

Reply:
I think we can just use `{rd} + {off}` here.

Reply:
ok, not exactly.